usb/util.py

```diff
@@ -254,22 +302,15 @@ def get_string(dev, index, langid = None):
     The return value is the unicode string present in the descriptor.
     """
     from usb.control import get_descriptor
+    langids = dev.langids
+
+    if 0 == len(langids):
+        raise ValueError("get_string from a device that offers none")
     if langid is None:
-	# Asking for the zero'th index is special - it returns a string
-	# descriptor that contains all the language IDs supported by the device.
-	# Typically there aren't many - often only one. The language IDs are 16
-	# bit numbers, and they start at the third byte in the descriptor. See
-	# USB 2.0 specification section 9.6.7 for more information.
-        #
-        # Note from libusb 1.0 sources (descriptor.c)
-        buf = get_descriptor(
-                    dev,
-                    254,
-                    DESC_TYPE_STRING,
-                    0
-                )
-        assert len(buf) >= 4
-        langid = buf[2] | (buf[3] << 8)
+        langid = langids[0]
+    else:
+        if langid not in langids:
+            raise ValueError("get_string in a language device does not support")
 
```

Comment:
This was based on my earlier comment above, but now it is irrelevant. Just replace the error string by "The device does not support the specified langid".